lib/locale.ts

```diff
@@ -3,11 +3,13 @@
  * SPDX-License-Identifier: GPL-3.0-or-later
  */
 
+const environmentLocale = Intl.DateTimeFormat().resolvedOptions().locale
+
 /**
  * Returns the user's locale
  */
 export function getLocale(): string {
-	return document.documentElement.dataset.locale || 'en'
+	return document.documentElement.dataset.locale || environmentLocale.replace(/-/g, '_')
```

Comment:
```suggestion
	return document.documentElement.dataset.locale || environmentLocale.replaceAll('-', '_')
```

maybe to make it clearer?

Reply:
👀 https://github.com/nextcloud-libraries/nextcloud-l10n/pull/864#discussion_r2058014142